src/hepstats/hypotests/calculators/asymptotic_calculator.py

```diff
@@ -125,35 +125,37 @@ def asimov_dataset(self, poi: POI):
             print(msg)
 
             self.set_params_to_bestfit()
-
             poiparam.floating = False
 
-            with poiparam.set_value(poivalue):
-                for trial in range(5):
-                    minimum = minimizer.minimize(loss=self.loss)
-                    if minimum.valid:
-                        break
+            if not self.loss.get_params():
+                values = {poiparam: {"value": poivalue}}
+
+            else:
+                with poiparam.set_value(poivalue):
+                    for trial in range(5):
```

Comment:
You could set this default of 5 as a function argument, just in case. (I did not look at the whole module so maybe that does not quite make sense)